pylint/checkers/variables.py

```diff
@@ -1469,7 +1469,11 @@ def _is_variable_violation(
                             nodes.Return,
                         ),
                     )
-                    and isinstance(defstmt.value, nodes.IfExp)
+                    and (
+                        isinstance(defstmt.value, nodes.IfExp)
+                        or isinstance(defstmt.value, nodes.Lambda)
```

Comment:
```suggestion
                        isinstance(defstmt.value, (nodes.IfExp, nodes.Lambda)),
```

Reply:
Doesn't `and` get precedence over `or`? I think this is checking for:
`IfExp` or (`Lambda` and `IfExp`).

Reply:
My bad !